tests/test_routes.py

```diff
@@ -159,6 +160,86 @@ def test_list_inventory_records(self):
         self.assertEqual(len(data), 5)
         self.assertCountEqual(expected_response, data)
 
+    # Test to update non-existent inventory records
+    def test_update_non_existent_inventory_records(self):
+        test_record = InventoryFactory()
+        response = self.client.post(BASE_URL, json=test_record.serialize())
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        logging.debug('Created record, %s', Inventory().deserialize(response.get_json()))
+        data = response.get_json()
+        # increment product_id so that query searches for a different product_id
+        data['product_id'] += 1
+        response = self.client.put("{}/{}/{}".format(BASE_URL, data['product_id'], data['condition']), json=data)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+    
+    # Test to update existing inventory records with random valid values
+    def test_update_inventory_records(self):
```

Comment:
Why do we need to have random over fields and values for checking updates?

Reply:
So that we can test for any possible combination of fields being updated at the same time.